runtime/src/snapshot_utils.rs

```diff
@@ -649,66 +901,188 @@ pub fn bank_from_snapshot_archive<P>(
 ) -> Result<(Bank, BankFromArchiveTimings)>
 where
     P: AsRef<Path> + std::marker::Sync,
+    Q: AsRef<Path> + std::marker::Sync,
```

Comment:
dumb question: so, `full_snapshot_archive_path` and `incremental_snapshot_archive_path` could be different types for any given this fn invocation? if possible, I want to avoid another type param (Q in this case)...

Reply:
In theory yes the types could be different, but in practice probably not. I forget which Rust resource I read that recommended this, but I did see it someone, so that one could be a `&Path` and the other could be a `PathBuf` and the function doesn't need to care about that. Without `Q`, then both FSS and ISS paths need to be the same type. Would you like me to remove `Q` here?

Reply:
@brooksprumo yeah, in general, fewer type params are desired to make code simpler if our current code base allows.

> I read that recommended this, but I did see it somewhere... So that one could be a &Path and the other could be a PathBuf and the function doesn't need to care about that.

seems you have good memory. I think this is perfectly good practice for library context. Unless sdk or related, our codebase is mostly application. so, we don't need to be that _generic_ :)

Reply:
Done!